Sources/MongoSwift/MongoCollection+Write.swift

```diff
@@ -21,24 +21,12 @@ extension MongoCollection {
      */
     @discardableResult
     public func insertOne(_ value: CollectionType, options: InsertOneOptions? = nil) throws -> InsertOneResult? {
-        let document = try self.encoder.encode(value).withID()
-        let opts = try self.encoder.encode(options)
-        var error = bson_error_t()
-        let reply = Document()
-        guard mongoc_collection_insert_one(self._collection, document.data, opts?.data, reply.data, &error) else {
-            throw getErrorFromReply(bsonError: error, from: reply)
+        do {
+            let result = try self.bulkWrite([InsertOneModel(value)], options: options?.asBulkWriteOpts())
```

Comment:
nit: I think there is inconsistency between creating the model before the `bulkWrite` call and passing it in, versus creating the model in the `bulkWrite` call itself. Slight preference for creating the model before, making it explicit.

Reply:
updated all to create the model(s) beforehand.